apisix/plugins/ai-proxy/schema.lua

```diff
@@ -0,0 +1,167 @@
+            default = 256
+
+        },
+        input_cost = {
+            type = "number",
+            description = "Defines the cost per 1M tokens in your prompt.",
+            minimum = 0
+
+        },
+        output_cost = {
+            type = "number",
+            description = "Defines the cost per 1M tokens in the output of the AI.",
+            minimum = 0
+
+        },
+        temperature = {
+            type = "number",
+            description = "Defines the matching temperature, if using chat or completion models.",
+            minimum = 0.0,
+            maximum = 5.0,
+
+        },
+        top_p = {
+            type = "number",
+            description = "Defines the top-p probability mass, if supported.",
+            minimum = 0,
+            maximum = 1,
+
+        },
+        upstream_host = {
+            type = "string",
+            description = "To be specified to override the host of the AI provider",
+        },
+        upstream_port = {
+            type = "integer",
+            description = "To be specified to override the AI provider port",
+
+        },
+        upstream_path = {
+            type = "string",
+            description = "To be specified to override the URL to the AI provider endpoints",
+        },
+        response_streaming = {
+            description = "Stream response by SSE",
+            type = "boolean",
+            default = false,
+        }
+    }
+}
+
+local model_schema = {
+    type = "object",
+    properties = {
+        provider = {
+            type = "string",
+            description = "AI provider request format - kapisix translates "
+                .. "requests to and from the specified backend compatible formats.",
+            oneOf = { "openai" }, -- add more providers later
+
+        },
+        name = {
+            type = "string",
+            description = "Model name to execute.",
+        },
+        options = model_options_schema,
+    },
+    required = {"provider", "name"}
+}
+
+_M.plugin_schema = {
+    type = "object",
+    properties = {
+        route_type = {
+            type = "string",
+            enum = { "llm/chat", "llm/completions", "passthrough" }
+        },
+        auth = auth_schema,
+        model = model_schema,
+    },
+    required = {"route_type", "model", "auth"}
+}
+
+_M.chat_request_schema = {
+    type = "object",
+    properties = {
+        messages = {
+            type = "array",
+            minItems = 1,
+            items = {
+                properties = {
+                    role = {
+                        type = "string",
+                        enum = {"system", "user", "assistant"}
+                    },
+                    content = {
+                        type = "string",
+                        minLength = "1",
+                    },
+                },
+                additionalProperties = false,
+                required = {"role", "content"},
+            },
+        }
```

Comment:
What design specifications do they follow? for OpenAI?
Does this mean that every new provider in the future will use this format?

Reply:
The current spec is for OpenAI, new providers can have different format which can be specified when they are being implemented.